mixer/adapter/kubernetes2/kubernetes.go

```diff
@@ -0,0 +1,409 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package kubernetes provides functionality to adapt mixer behavior to the
+// kubernetes environment. Primarily, it is used to generate values as part
+// of Mixer's attribute generation preprocessing phase. These values will be
+// transformed into attributes that can be used for subsequent config
+// resolution and adapter dispatch and execution.
+package kubernetes
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	"k8s.io/api/core/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth" // needed for auth
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"context"
+
+	"istio.io/istio/mixer/adapter/kubernetes2/config"
+	kubernetes_apa_tmpl "istio.io/istio/mixer/adapter/kubernetes2/template"
```

Comment:
nit: make the import name much shorter. something like: `ktmpl` or `k8stmpl`.

Reply:
D